### ISSUE TITLE
Add a "test output" command

### ISSUE THREAD

**Issue by AndersonQ** [elastic/elastic-agent#2614]
**Describe the enhancement:**

Add a command for the agent which would test all the connections the agent make:
 - fleet-server
 - elasticsearch or other output
 - artifacts-api

The output of the command would be the config used for each connection such as the address, proxy, any special SSL setting and so on, and the result of the request, the HTTP status and any error, if any.

Similar to the beats `test output` command.

It could also be part of the diagnostic bundle.

Another further enhancement would be to have traceroute like information, there are a few Go implementations for that:
 - https://github.com/aeden/traceroute
 - https://github.com/pixelbender/go-traceroute

**Describe a specific use case for the enhancement or feature:**

As the agent accepts to configure different proxies for fleet-serve, ES output and a global one, and our users have all sorts of network setup, its often questioned if the proxy or any other connection config was properly done. Having such a command would help to identify any issues.

** What is the definition of done?**

an installed agent on any OS successfully runs a `test output` command and the information provided clearly shows how the connection to fleet-server, ES (or other) output and artefacts-api id done, which configs are used, and if those connections are successfully established, or the error, if any.

**Comment by cmacknz** [comment]
One complexity to this is that the agent is a supervisor. Even if the agent process itself can reach the output with the configuration provided, that does not guarantee that each of the sub-processes can with the same configuration. It could be possible for the agent to be able to reach Fleet or Elasticsearch but not Elastic Defend.

**Comment by cmacknz** [comment]
I should note that today Elastic Defend contacts each of Fleet Server, the Artifacts API, and the data output (Elasticsearch, Logstash, etc) independently of the agent.

**Comment by AndersonQ** [comment]
> One complexity to this is that the agent is a supervisor. Even if the agent process itself can reach the output with the configuration provided, that does not guarantee that each of the sub-processes can with the same configuration. It could be possible for the agent to be able to reach Fleet or Elasticsearch but not Elastic Defend.

Agreed, ideally all the components would do the same, perhaps the agent's `test output` could take a flag `--include-components`` and the agent would request the components to do the same test.

Of course it either means the agent would call it on every component or send an action

**Comment by richlv** [comment]
This is a very helpful feature in filebeat, and either doing it manually for components or adding the suggested `--include-components` seem like reasonable approaches.
